matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/SessionModule.kt

```diff
@@ -285,6 +289,19 @@ internal abstract class SessionModule {
         fun providesMxCryptoConfig(matrixConfiguration: MatrixConfiguration): MXCryptoConfig {
             return matrixConfiguration.cryptoConfig
         }
+
+        @JvmStatic
+        @Provides
+        @SessionScope
+        fun providesAuthApi(
+                @Unauthenticated okHttpClient: Lazy<OkHttpClient>,
+                retrofitFactory: RetrofitFactory,
+                homeServerConnectionConfig: HomeServerConnectionConfig
+        ): AuthAPI {
+            val homeServerUrl = homeServerConnectionConfig.homeServerUriBase.toString()
+            return retrofitFactory.create(okHttpClient, homeServerUrl)
+                    .create(AuthAPI::class.java)
+        }
```

Comment:
I do not really like to add a provider for `AuthAPI`. It breaks a bit the arch of login/registration flow.

I would create a new `RefreshTokenAPI` interface, that can be created using an injected Retrofit instance (like [here](https://github.com/vector-im/element-android/blob/develop/matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/RoomModule.kt#L133) for instance)